src/saml.py

```diff
@@ -66,6 +77,21 @@ def metadata_url(self) -> str:
         """
         return self._charm_state.metadata_url
 
+    @cached_property
+    def _is_certificate_valid(self):
+        """Validate the public certificate for the metadata URL.
+
+        Returns:
+            True if the certificate matches the one provided.
+        """
+        if self._charm_state.certificate:
+            url = urllib.parse.urlparse(self._charm_state.metadata_url)
+            certificate = ssl.get_server_certificate(
+                (url.hostname, url.port if url.port else 443), timeout=10
```

Comment:
Is the SAML certificate guarantee to be the same as the TLS certificate of the metadata website?

Reply:
No, the requirement is to be able to validate the URL's certfiicate. The metadata might define one or more certificates for the SP to use. Do you think this is unclear?

Reply:
I think the most conventional way to verify or pin a certificate is to let user provide a fingerprint, and verify the certificate against the fingerprint. And if the user want to match the TLS certificate, they can just provide the fingerprint of the TLS certificate, which is very easy to obtain in the browser.

Reply:
Sorry I misunderstood your intension here. But I still think the fingerprint could be a better alternative to comparing certificate directly.

Reply:
Done! Thx for the feedback